Mathlib/Order/LatticeIntervals.lean

```diff
@@ -173,6 +211,19 @@ theorem coe_top [Preorder α] [OrderTop α] {a : α} : ↑(⊤ : Ici a) = (⊤ :
 instance boundedOrder [Preorder α] [OrderTop α] {a : α} : BoundedOrder (Ici a) :=
   { Ici.orderTop, Ici.orderBot with }
 
+
+protected lemma disjoint_iff [SemilatticeInf α] {a : α} {x y : Ici a} :
+    Disjoint x y ↔ ↑x ⊓ ↑y = a := by
+  simp [_root_.disjoint_iff, Subtype.ext_iff]
+
+protected lemma codisjoint_iff [SemilatticeSup α] [OrderTop α] {a : α} {x y : Ici a} :
+    Codisjoint x y ↔ Codisjoint (x : α) (y : α) := by
```

Comment:
I'd expect this lemma to be stated the other way, or at to be `norm_cast`

Reply:
This matches the existing API, and `norm_cast` is illegal here.

Reply:
Sorry, my message got mangled. I meant "be `norm_cast` the other way"

Reply:
I think in most cases the current forward direction is more convenient, so I'd prefer to leave it as is. `norm_cast ←` might be a nice feature to request in core.